### PR TITLE
[NO-TICKET] Token code refactoring and documentation

### PR DESCRIPTION
**Pull request by forrestbaer**
## Summary

- Moved utilities that had dependencies (node, fs) to a new file so compiled token modules can more easily be used with other tools (react-sketchapp in particular)
- Updated `hexOpacity` to use float values instead of integers to be more consistent with normal css
- Several refactors to simplify code and several variable name changes for easier comprehension
- Added TSDoc/TypeDoc style documentation for future extraction
- Added missing return types in components, functions and themes

## How to test

in root: `yarn token:dist` `yarn build` `yarn loki test` (for each subsystem if you want, they all pass)
in design-system-tokens: `yarn build tokens scss && yarn build tokens csv && yarn build tokens json && yarn build themes scss && yarn build themes csv && yarn build themes json`

### CODE REVIEW

File: packages/design-system-tokens/src/lib/types.ts [hunk 1]
```diff
@@ -32,23 +32,13 @@ export type SpacerTokens = Token<MeasureValues>;
...
-  | BorderRadiusTokens
-  | ColorTokens
-  | FontTokens
-  | MeasureTokens
-  | MediaWidthTokens
-  | ShadowTokens
-  | SpacerTokens
-  | TimeTokens
-  | zIndexTokens;
+export type AnyTokenValues = Token<any>;
```
Comment (pwolfert) [inline packages/design-system-tokens/src/lib/types.ts:35]: What's the reason for making this less restrictive? Did a token not fit inside one of the categories?

File: packages/design-system-tokens/sketch/cmsds-token-importer/src/json-token-importer.js [hunk 2]
```diff
@@ -36,71 +21,6 @@ const makeColorSwatches = (colorTokens) => {
   return swatches;
 };
 
-/*
- * create layer styles based on component values
- * TODO: when first set of layer stlyes (buttons) are created
- * this can be fully tested out & completed
- */
-const makeLayerStyles = (componentVals) => {
```
[the remark this answers is not in the document]
Comment (forrestbaer) [reply packages/design-system-tokens/sketch/cmsds-token-importer/src/json-token-importer.js:44]: These were placeholders that are no longer necessary. Forgot to remove them below!

File: packages/design-system-tokens/src/lib/types.ts (hunk 1 above)
Comment (forrestbaer) [reply packages/design-system-tokens/src/lib/types.ts:35]: The places they were being used needed some more flexibility yeah, I'm not sure how much at the moment so can be revised in the future.

File: packages/design-system-tokens/sketch/cmsds-token-importer/src/json-token-importer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (forrestbaer) [reply packages/design-system-tokens/sketch/cmsds-token-importer/src/json-token-importer.js:44]: Removed the references below, when the time comes to add them back the file history can be referenced.

**Review by pwolfert — APPROVED** [review]

**Pull request merged**